src/history/html5.js

```diff
@@ -52,6 +52,12 @@ export class HTML5History extends History {
 
   replace (location: RawLocation, onComplete?: Function, onAbort?: Function) {
     const { current: fromRoute } = this
+    if (typeof location === 'string') {
+      location = { path: location }
+    }
+    if (typeof location === 'object' && !location.replace) {
+      (location: Object).replace = true
+    }
```

Comment:
Since location is an object at this point, we can directly set `replace` to true

```suggestion
  (location: Object).replace = true
```